src/java/com/puppetlabs/http/client/impl/JavaClient.java

```diff
@@ -324,27 +337,52 @@ public static void requestWithClient(final RequestOptions requestOptions,
 
         final HttpContext httpContext = HttpClientContext.create();
 
-        final FutureCallback<HttpResponse> futureCallback = new FutureCallback<HttpResponse>() {
+        final Timer.Context timerContext = registry == null ? null : timer(registry, request).time();
```

Comment:
I think I like that. I think not needing the separate `timedXxx()` methods is a lot cleaner. I'll play with it a bit tomorrow. Thanks!